src/mlpack/core/optimizers/cmaes/cmaes_impl.hpp

```diff
@@ -0,0 +1,781 @@
+
+      // update covariance matrix
+      for (int i = 0; i < N; ++i)
+        for (int j = diag ? i : 0; j <= i; ++j)
+        {
+          double& Cij = C(i,j);
+          Cij = onemccov1ccovmu*Cij + ccov1 * (pc[i]*pc[j] + longFactor*Cij);
+          for (int k = 0; k < mu; ++k)
+          { // additional rank mu update
+            Cij += ccovmu*weights[k] * (population(index[k] , i) - xold[i])
+                * (population(index[k] , j) - xold[j]) / sigmasquare;
+          }
+        }
+      // update maximal and minimal diagonal value
+      maxdiagC = arma::max(C.diag());
+      mindiagC = arma::min(C.diag());
+     
+    }
+  }
+
+
+  /**
+   * Adds the mutation sigma*B*(D*z).
+   * @param x Search space vector.
+   * @param eps Mutation factor.
+   */
+  template<typename funcType>
+  void CMAES<funcType>::addMutation(double* x, double eps)
+  {
+    for (int i = 0; i < N; ++i)
+      tempRandom[i] = rgD[i]*rand.gauss();
+    for (int i = 0; i < N; ++i)
+    {
+      double sum = 0.0;
+      for (int j = 0; j < N; ++j)
+        sum += B(i,j)*tempRandom[j];
+      x[i] = xmean[i] + eps*sigma*sum;
+    }
+  }
+
+  /**
+   * Initializes the CMA-ES algorithm.
+   * @param parameters The CMA-ES parameters in the parameters.h file
+   * @return Array of size lambda that can be used to assign fitness values and
+   *         pass them to updateDistribution()
+   */
+  template<typename funcType>
+  void CMAES<funcType>::init(arma::vec& func)
+  {
+
+    stopMessage = "";
+
+    double trace = arma::accu(arma::pow(rgInitialStds, 2));
+    sigma = std::sqrt(trace/N);
+
+    chiN = std::sqrt((double) N) * (1 - 1/(4*N) + 1/(21*N*N));
+    eigensysIsUptodate = true;
+    doCheckEigen = false;
+    genOfEigensysUpdate = 0;
+
+    double dtest;
+    for (dtest = double(1); dtest && dtest < double(1.1)*dtest; dtest *= double(2))
+      if (dtest == dtest + double(1))
+        break;
+    dMaxSignifKond = dtest / double(1000);
+
+    gen = 0;
+    countevals = 0;
+    state = INITIALIZED;
+    dLastMinEWgroesserNull = double(1);
+
+    pc.set_size(N);
+    ps.set_size(N);
+    tempRandom.set_size(N+1);
+    BDz.set_size(N);
+    xmean.set_size(N+2);
+    xmean[0] = N;
+    ++xmean;
+    xold.set_size(N+2);
+    xold[0] = N;
+    ++xold;
+    xBestEver.set_size(N+3);
+    xBestEver[0] = N;
+    ++xBestEver;
+    xBestEver[N] = std::numeric_limits<double>::max();
+    rgD.set_size(N);
+    C.set_size(N,N);
+    B.set_size(N,N);
+    publicFitness.set_size(lambda);
+    functionValues = new double[lambda+1];
+    functionValues[0] = lambda;
+    ++functionValues;
+    const int historySize = 10 + (int) ceil(3.*10.*N/lambda);
+    funcValueHistory = new double[historySize + 1];
+    funcValueHistory[0] = (double) historySize;
+    funcValueHistory++;
+    index = arma::linspace<arma::vec>(0, lambda-1, lambda);
+    population.zeros(lambda, N+2);
+    for (int i = 0; i < lambda; i++)
+    {
+      functionValues[i] = std::numeric_limits<double>::max();
```

Comment:
Please use ``arma::vec`` here and you can do: ``functionValues.fill(DBL_MAX)`` to do the same without a for loop.